### ISSUE TITLE
Add support for TRT İzle

### ISSUE THREAD

**Issue by ahmetlii** [ytdl-org/youtube-dl#30906]
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://www.trtizle.com/belgesel/dunya-tarihinin-donum-noktalari/dunya-tarihinin-donum-noktalari-1-bolum-125583 (requires signing in)
- Live stream of the channel: https://www.trtizle.com/canli/tv/trt-2 (do not require signing in)
- Playlist: https://www.trtizle.com/listem (requires signing in)

## Description

This is the state television company of Turkey, and therefore it's a legal site. However, watching the episodes of programs or making a private list require signing in. Obtaining account credentials is simple: register on the site with an email.

**Comment by dirkf** [comment]
In the first example, a manifest and thumbnail is found in the `<figure class="show-detail--image hasVideo">` element, without needing to log in.

The livestream apparently needs to be traced with JS enabled to reveal the stream URLs.

The playlist needs a login. Am I right to guess that personal lists are behind the login?

There are also Programs URLs, eg the master list of shows https://www.trtizle.com/programlar, a list for one show (eg https://www.trtizle.com/belgesel/dunya-tarihinin-donum-noktalari), ...

The program page for "Dünya Tarihinin Dönüm Noktaları" looks very similar to the page for episode 1 (which has `/dunya-tarihinin-donum-noktalari-1-bolum-125583` added to the URL), so there should be a separate playlist extractor for that; the episode list is in the page.






In the

**Comment by ahmetlii** [comment]
@dirkf 
> The playlist needs a login. Am I right to guess that personal lists are behind the login?
Correct, and as far as I am aware it doesn't have a feature to share lists (hence there also isn't public lists).
Also I've just realized there are some other master lists I've just noticed (other than documentaries and programs): https://www.trtizle.com/engelsiz and https://www.trtizle.com/diziler
And yes, while viewing the episode page without signing in is same as the program page, it directly shows the episode in the video player on episode page (hence the almost exact screen that can be seen in the livestream pages), and nothing else.
